### PR TITLE
[Components with more than 1 service] LPS-187673 Refactor ManagedServiceFactory components from multiple modules

### PR DESCRIPTION
**Pull request by RenanUmV**
[LPS-187673](https://liferay.atlassian.net/browse/LPS-187673)

### CODE REVIEW

**Comment by RenanUmV** [comment]
ci:test:startup

**Comment by RenanUmV** [comment]
ci:test:lima

**Comment by RenanUmV** [comment]
ci:test:echo

**Comment by RenanUmV** [comment]
ci:test:security

**Review by GabrielHenriquedev — APPROVED** [review]

**Comment by dantewang** [comment]
@RenanUmV Converted to Draft as we are in discussion.

**Comment by RenanUmV** [comment]
ci:test:sf

**Comment by RenanUmV** [comment]
ci:test:sf

**Comment by RenanUmV** [comment]
ci:test:relevant

**Comment by RenanUmV** [comment]
ci:test:sf

**Comment by RenanUmV** [comment]
ci:test:sf

**Comment by RenanUmV** [comment]
ci:test:relevant

**Comment by RenanUmV** [comment]
ci:test:sf

File: modules/apps/redirect/redirect-service/src/main/java/com/liferay/redirect/internal/configuration/RedirectURLSettingsImpl.java [hunk 1]
```diff
@@ -5,46 +5,141 @@
...
-import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Modified;
 
 /**
  * @author Drew Brokke
  */
-@Component(service = RedirectURLSettings.class)
+@Component(
+	configurationPid = "com.liferay.redirect.internal.configuration.RedirectURLConfiguration",
```
Comment (rafaprax) [inline modules/apps/redirect/redirect-service/src/main/java/com/liferay/redirect/internal/configuration/RedirectURLSettingsImpl.java:32]: Hey @RenanUmV, Why did you copy the `RedirectURLManagedServiceFactory` logic to here? Please check if it is needed. Thx

**Review by rafaprax — CHANGES_REQUESTED** [review]

**Review by rafaprax — CHANGES_REQUESTED** [review]

File: modules/apps/redirect/redirect-service/src/main/java/com/liferay/redirect/internal/configuration/RedirectURLSettingsImpl.java (hunk 1 above)
Comment (RenanUmV) [reply modules/apps/redirect/redirect-service/src/main/java/com/liferay/redirect/internal/configuration/RedirectURLSettingsImpl.java:32]: I forgot to delete the RedirectURLManagedServiceFactory class

**Comment by RenanUmV** [comment]
ci:test:sf

File: modules/apps/oauth2-provider/oauth2-provider-rest/src/main/java/com/liferay/oauth2/provider/rest/internal/endpoint/access/token/grant/handler/LiferayJWTBearerGrantHandler.java [hunk 2]
```diff
@@ -58,9 +75,27 @@ public List<String> getSupportedGrantTypes() {
 	}
 
 	@Activate
-	protected void activate(Map<String, Object> properties) {
+	protected void activate(
+		BundleContext bundleContext, Map<String, Object> properties) {
+
 		_oAuth2ProviderConfiguration = ConfigurableUtil.createConfigurable(
 			OAuth2ProviderConfiguration.class, properties);
+
```
Comment (rafaprax) [inline modules/apps/oauth2-provider/oauth2-provider-rest/src/main/java/com/liferay/oauth2/provider/rest/internal/endpoint/access/token/grant/handler/LiferayJWTBearerGrantHandler.java:83]: Hey @RenanUmV, please move the `_oAuth2ProviderConfiguration` initialization to the `modified` method. Thx

File: modules/apps/document-library/document-library-preview-pdf/src/main/java/com/liferay/document/library/preview/pdf/internal/configuration/admin/service/helper/PDFPreviewManagedServiceFactoryHelper.java [hunk 3]
```diff
@@ -0,0 +1,371 @@
...
+import org.osgi.service.cm.ConfigurationAdmin;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Joao Victor Alves
+ */
+@Component(service = PDFPreviewManagedServiceFactoryHelper.class)
+public class PDFPreviewManagedServiceFactoryHelper {
+
```
Comment (rafaprax) [inline modules/apps/document-library/document-library-preview-pdf/src/main/java/com/liferay/document/library/preview/pdf/internal/configuration/admin/service/helper/PDFPreviewManagedServiceFactoryHelper.java:34]: @RenanUmV, please rename to `PDFPreviewConfigurationHelper`

File: modules/apps/portal-security-sso/portal-security-sso-openid-connect-impl/src/main/java/com/liferay/portal/security/sso/openid/connect/internal/configuration/admin/service/OpenIdConnectProviderEveryNodeEveryStartup.java [hunk 4]
```diff
@@ -57,43 +62,13 @@
...
-		ManagedServiceFactory.class,
-		OpenIdConnectProviderManagedServiceFactory.class,
-		PortalInstanceLifecycleListener.class
-	}
+	configurationPid = "com.liferay.portal.security.sso.openid.connect.internal.configuration.OpenIdConnectProviderConfiguration",
+	service = PortalInstanceLifecycleListener.class
 )
 @Deprecated
-public class OpenIdConnectProviderManagedServiceFactory
+public class OpenIdConnectProviderEveryNodeEveryStartup
```
Comment (rafaprax) [inline modules/apps/portal-security-sso/portal-security-sso-openid-connect-impl/src/main/java/com/liferay/portal/security/sso/openid/connect/internal/configuration/admin/service/OpenIdConnectProviderEveryNodeEveryStartup.java:69]: @RenanUmV, please rename to `OpenIdConnectProviderPortalInstanceLifecycleListener`. Thx

**Review by rafaprax — CHANGES_REQUESTED** [review]

**Pull request closed without merging**